### PR TITLE
Updating permissions.py to accept dot and dop tokens

### PR DESCRIPTION
**Pull request by vad33r**

### CODE REVIEW

File: eox_core/api/v1/permissions.py [hunk 1]
```diff
@@ -40,7 +40,11 @@ def has_permission(self, request, view):
         To grant access, checks if the requesting user either can call eox-core API or if it's an admin user.
         """
         try:
-            allowed_for_site = request.user.is_staff or request.get_host() in request.auth.client.url
+            if request.user.is_staff:
+                allowed_for_site = request.user.is_staff
+            else:
+                allowed_for_site = request.auth.application.redirect_uri_allowed(request.build_absolute_uri('/')) or \
```
Comment (andrey-canon) [inline eox_core/api/v1/permissions.py:46]: This always raises an exception, doesn't it ? when the request is with dop

File: eox_core/api/v1/permissions.py [hunk 2]
```diff
@@ -40,7 +40,11 @@ def has_permission(self, request, view):
         To grant access, checks if the requesting user either can call eox-core API or if it's an admin user.
         """
         try:
-            allowed_for_site = request.user.is_staff or request.get_host() in request.auth.client.url
+            if request.user.is_staff:
+                allowed_for_site = request.user.is_staff
```
Comment (felipemontoya) [inline eox_core/api/v1/permissions.py:44]: This first part of the if is just confusing. If the is in fact staff we are making the whole function take a different path. My suggestion would be to completely remove from the logic by looking that up first and returning.
If the user is not staff, then you can do the auth.application or auth.client checks each in a try..catch block

**Review by felipemontoya — CHANGES_REQUESTED** [review]
Please remember to add the unit tests for this changeset

File: eox_core/api/v1/permissions.py (hunk 1 above)
Comment (vad33r) [reply eox_core/api/v1/permissions.py:46]: No, when the request is with dop tries with:
`request.get_host() in request.auth.client.url`

**Comment by mariajgrimaldi** [comment]
I know the title of the PR is self-explanatory but can you add a description? That way if someone returns to this PR they can have more context.

File: eox_core/api/v1/permissions.py [hunk 3]
```diff
@@ -39,8 +39,14 @@ def has_permission(self, request, view):
         """
         To grant access, checks if the requesting user either can call eox-core API or if it's an admin user.
         """
+        if request.user.is_staff:
+            return request.user.is_staff
         try:
-            allowed_for_site = request.user.is_staff or request.get_host() in request.auth.client.url
+            allowed_for_site = request.get_host() in request.auth.client.url
```
Comment (andrey-canon) [inline eox_core/api/v1/permissions.py:45]: if this is True, the line 49 will raise the exception and do this False ...

File: eox_core/api/v1/permissions.py [hunk 4]
```diff
@@ -39,8 +39,17 @@ def has_permission(self, request, view):
...
         try:
-            allowed_for_site = request.user.is_staff or request.get_host() in request.auth.client.url
+            allowed_for_site = request.auth.application.redirect_uri_allowed(request.build_absolute_uri('/'))
+            return request.user.has_perm('auth.can_call_eox_core')
+        except Exception:  # pylint: disable=broad-except
+            allowed_for_site = False
+
+        try:
+            allowed_for_site = request.get_host() in request.auth.client.url
+            return request.user.has_perm('auth.can_call_eox_core')
```
Comment (andrey-canon) [inline eox_core/api/v1/permissions.py:52]:
```suggestion
            if hasattr(request.auth, 'application'):
                allowed_for_site = request.auth.application.redirect_uri_allowed(request.build_absolute_uri('/'))
            else:
                allowed_for_site = request.get_host() in request.auth.client.url
```

Comment (vad33r) [reply eox_core/api/v1/permissions.py:52]: Ok, I didn't know what I could do `if hasattr(request.auth, 'application'):` ty

**Review by andrey-canon — COMMENTED** [review]
@vad33r  did you test these last changes ??

**Comment by vad33r** [comment]
Yes, i tested them

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**